### PR TITLE
fix apex sha256 and depends constraint

### PR DESCRIPTION
**Pull request by darmac**

### CODE REVIEW

**Comment by darmac** [comment]
The latest apex version need some error fix accroding to this issue:
https://github.com/khuck/xpress-apex/issues/125

File: var/spack/repos/builtin/packages/apex/package.py [hunk 1]
```diff
@@ -12,9 +12,9 @@ class Apex(CMakePackage):
     homepage = "http://github.com/khuck/xpress-apex"
     url      = "http://github.com/khuck/xpress-apex/archive/v0.1.tar.gz"
 
-    version('0.1', sha256='bb0be37f8f8133fe492998515bcf66a4df452c28a995d317228fbed9b18e6a92')
+    version('0.1',   sha256='efd10f38a61ebdb9f8adee9dc84acaee22d065b2e6eea1888872a7bfca0f4495')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/apex/package.py:15]: It looks like there are dozens of newer releases. Can you contact the developers and find out why the checksum changed?

**Comment by khuck** [comment]
No idea why the hash was changed, if I changed it, that was a long time ago.  I haven't kept the package up to date with the releases, I should probably do that this week.  I wasn't aware anyone was using it other than me, and I always used the develop build.  Thanks!

Actually, looking at this change, it's not quite correct. The latest version of APEX works with all binutils releases, but the previous ones don't, due to the API change in binutils 2.34.

**Comment by darmac** [comment]
@khuck 
Got it, thanks for your suggestion.
I will update apex version and dependants after the compile error fixed.

**Comment by khuck** [comment]
No worries, I need to fix many other issues, so I will be updating the package.py this week, too.  Thanks!

**Review by adamjstewart — APPROVED** [review]

**Comment by adamjstewart** [comment]
I'm going to merge this and let @khuck take care of the rest.

@khuck if you add:
```python
maintainers = ['khuck']
```
to the `package.py`, it will give us someone to ping when a user reports a build issue or submits a PR to modify the package.

**Pull request merged**